unic-locale-impl/src/extensions/transform.rs

```diff
@@ -72,6 +151,29 @@ impl TransformExtensionList {
         Ok(())
     }
 
+    /// Removes a tfield from the `TransformExtensionList`.
+    ///
+    /// Returns `true` if tfield was included in the `TransformExtensionList`
+    /// before removal.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use unic_locale_impl::Locale;
+    ///
+    /// let mut lo: Locale = "en-US-t-k0-dvorak".parse()
+    ///     .expect("Parsing failed.");
+    ///
+    /// assert_eq!(lo.extensions.transform.remove_tfield("k0")
+    ///                .expect("Removing tfield failed."),
+    ///            true);
+    ///
+    /// assert_eq!(lo.to_string(), "en-US");
+    /// ```
+    pub fn remove_tfield(&mut self, tkey: &str) -> Result<bool, LocaleError> {
+        Ok(self.tfields.remove(&parse_tkey(tkey)?).is_some())
+    }
+
```

Comment:
Add `clear_tfields` and `get_tfield_keys` which returns `impl ExactSizeIterator`.

Reply:
Leaving `get_tfield_keys` for separate commit.